<launch>
    <arg name="targetDeviceId" default="aDevice" ></arg>
    <arg name="rangingTopic" default="/gtec/toa/ranging" ></arg>
    <arg name="anchorsTopic" default="/gtec/toa/anchors" ></arg>
    <arg name="toaTagId" default="0" ></arg>
    <arg name="useStartPosition" default="0" ></arg>
    <arg name="initAngle" default="0.0" ></arg>
    <arg name="initPositionX" default="0.0" ></arg>
    <arg name="initPositionY" default="0.0" ></arg>
    <arg name="initPositionZ" default="0.0" ></arg>
    <arg name="use2d" default="1" ></arg>
    <arg name="variant" default="0" ></arg>
    <arg name="numRangingsToIgnore" default="0" ></arg>

    <node name="ml_toa_$(arg targetDeviceId)" output="screen" pkg="gtec_roskfpos" type="kfpos" launch-prefix="">
        <param name="rate" type="int" value="20" />
        <param name="targetDeviceId" type="string" value="$(arg targetDeviceId)"></param>
        <param name="rangingTopic" type="string" value="$(arg rangingTopic)"></param>
        <param name="anchorsTopic" type="string" value="$(arg anchorsTopic)"></param>
        <param name="tagId" type="string" value="$(arg toaTagId)"></param>
        <param name="algorithm" type="string" value="ALGORITHM_ML"></param>
        <param name="useStartPosition" type="int" value="$(arg useStartPosition)" />
        <param name="initAngle" type="double" value="$(arg initAngle)" />
        <param name="initPositionX" type="double" value="$(arg initPositionX)" />
        <param name="initPositionY" type="double" value="$(arg initPositionY)" />
        <param name="initPositionZ" type="double" value="$(arg initPositionZ)" />
        <param name="use2d" type="int" value="$(arg use2d)" />
        <param name="variant" type="int" value="$(arg variant)" />
        <param name="numRangingsToIgnore" type="int" value="$(arg numRangingsToIgnore)" />
    </node>
</launch>

